### ISSUE TITLE
Neovim multi key input not work expect with Chinese input method.

### ISSUE THREAD

**Issue by fishioon** [alacritty/alacritty#8113]
Chinese input method not work with neovim, but iterm2 work fine.
When input one key, like "j", work fine. when input multi key，like "jjjj", not work expect.

https://github.com/user-attachments/assets/ef9085d0-a2af-42e0-b1be-db9b5b310bfc

### System

OS: macOS
Version: alacritty 0.13.2 (bb8ea18)

### Logs

0.000010713s] [INFO ] [alacritty] Welcome to Alacritty
[0.000404124s] [INFO ] [alacritty] Version 0.13.2 (bb8ea18)
[0.001080646s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/Users/fish/.config/alacritty/alacritty.toml"
[0.001713901s] [DEBUG] [alacritty] Using environment locale: en_US.UTF-8
[0.034580350s] [INFO ] [alacritty] Using Apple CGL
[0.059139955s] [DEBUG] [alacritty] Picked GL Config:
                                     buffer_type: Some(Rgb { r_size: 8, g_size: 8, b_size: 8 })
                                     alpha_size: 8
                                     num_samples: 0
                                     hardware_accelerated: true
                                     supports_transparency: Some(true)
                                     config_api: Api(OPENGL)
                                     srgb_capable: true
[0.100830860s] [INFO ] [alacritty] Window scale factor: 2
[0.101910826s] [DEBUG] [alacritty] Loading "MonaspiceNe Nerd Font Mono" font
[0.185876642s] [INFO ] [alacritty] Running on AMD Radeon Pro 5300M OpenGL Engine
[0.185952566s] [INFO ] [alacritty] OpenGL version 4.1 ATI-5.5.17, shader_version 4.10
[0.185982217s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.194738398s] [DEBUG] [alacritty] Filling glyph cache with common glyphs
[0.221958434s] [INFO ] [alacritty] Cell size: 22 x 43
[0.222053454s] [INFO ] [alacritty] Padding: 8 x 17

**Comment by chrisduerr** [comment]
Your video does not work.

**Comment by chrisduerr** [comment]
Is this only an issue in neovim? It should have nothing to do with this.

**Comment by kchibisov** [comment]
Probably because we insert the IME text as bracketed text when it's a _string_ and not a single character?

generally though, it's strange that IME makes all of that pending.

**Comment by chrisduerr** [comment]
> Probably because we insert the IME text as bracketed text when it's a string and not a single character?

Yeah but I'd argue that's probably correct behavior? Why would inserting `ff` or whatever as one IME string trigger shortcuts?

**Comment by fishioon** [comment]
> > Probably because we insert the IME text as bracketed text when it's a string and not a single character?
> 
> Yeah but I'd argue that's probably correct behavior? Why would inserting `ff` or whatever as one IME string trigger shortcuts?

Sometimes I forget to switch to the English input method and press jj or kk Neovim move shortcuts, only then realizing that the input method was not switched. Even if I use the Ctrl+Space shortcut to switch to the English input method, the previously input jj or kk still written into Neovim.

**Comment by chrisduerr** [comment]
I'll leave this up to @kchibisov, I don't care about IME myself.

**Comment by CG-man** [comment]
This feature worked as expected in version 12, but it stopped working when I updated it recently

**Comment by cyphbt** [comment]
You can use im-select (https://github.com/daipeihust/im-select)
